chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -513,6 +513,20 @@ impl Ledger {
                     check::valid_block0_cert_transaction(&tx)?;
                     ledger = ledger.mint_token_unchecked(tx.payload().into_payload())?;
                 }
+                Fragment::Evm(tx) => {
+                    #[cfg(feature = "evm")]
+                    {
+                        // WIP: deploying contract
+                        let contract = tx.as_slice().payload().into_payload();
```

Comment:
We should check this is a valid Jormungandr transaction anyway (i.e. ios are balanced, limits are respected, etc), even if we set the fees to zero

Reply:
The plan here is that this is going to be not a Jormungandr transaction, but something living in its own EVM world. Not even a certificate.